core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -35,11 +36,31 @@ public KafkaLocationManager(final ZooKeeperHolder zkFactory, final KafkaSettings
         this.zkFactory = zkFactory;
         this.kafkaProperties = new Properties();
         this.kafkaSettings = kafkaSettings;
+        applySecurityProperties();
         this.updateBootstrapServers(true);
         this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
         this.scheduledExecutor.scheduleAtFixedRate(() -> updateBootstrapServersSafe(false), 1, 1, TimeUnit.MINUTES);
     }
 
+    private void applySecurityProperties() {
+        if (this.kafkaSettings.getSecurityProtocol().isPresent()) {
+            this.kafkaProperties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+                    this.kafkaSettings.getSecurityProtocol().get());
+        }
+        if (this.kafkaSettings.getSaslMechanism().isPresent()) {
+            this.kafkaProperties.put(SaslConfigs.SASL_MECHANISM,
+                    this.kafkaSettings.getSaslMechanism().get());
+        }
+        if (this.kafkaSettings.getKafkaUsername().isPresent() &&
+                this.kafkaSettings.getKafkaPassword().isPresent()) {
+            this.kafkaProperties.put(SaslConfigs.SASL_JAAS_CONFIG,
+                    "org.apache.kafka.common.security.plain.PlainLoginModule required " +
+                            "username=" + this.kafkaSettings.getKafkaUsername().get()
+                            + " password=\"" + this.kafkaSettings.getKafkaPassword().get() + "\";");
+        }
+        LOG.info("Kafka security settings" + this.kafkaProperties.toString());
```

Comment:
please, do not set user/password to the logs

Reply:
Ouch